### PR TITLE
Implement the refund modal

### PR DESCRIPTION
**Pull request by deivamagalhaes**
# Summary

This PRs implements the refund modal.

## UI Changes

- Refund modal: https://cloud.skyver.ge/4guJwE7p

### Setup

- Have a commerce order (you can place one through Facebook or change the `_created_via` meta for one to `facebook`

### Steps

1. Edit the commerce order
1. Set the Status to Refunded
1. Click Update
    - [x] The refund modal is displayed
    - [x] The refund modal contains a select for the refund reason
1. Click Cancel
    - [x] The form is not submitted
1. Click Update again
    - [x] The refund modal is displayed
    - [x] The refund modal contains a select for the refund reason
1. Select a refund reason
1. Click Submit refund
    - [x] The form is submitted

## Before merge

- [x] I have confirmed these changes in each supported minor WooCommerce version

### CODE REVIEW

**Review by unfulvio — COMMENTED** [review]
hey @deivamagalhaes I'm following the QA steps but for some reason this doesn't look to be working... I am pretty sure my order created via is either `facebook` or `instagram`; I tried changing the status to refunded either from the order status dropdown or the `Refund` button in the order items box.

**Comment by adrianostanley** [comment]
> hey @deivamagalhaes I'm following the QA steps but for some reason this doesn't look to be working... I am pretty sure my order created via is either `facebook` or `instagram`; I tried changing the status to refunded either from the order status dropdown or the `Refund` button in the order items box.

@unfulvio I believe you're having the same issue as me... by having the following filters...

```
  add_filter( 'wc_facebook_commerce_is_available',   '__return_true' );
  add_filter( 'wc_facebook_commerce_is_connected',   '__return_true' );
  add_filter( 'wc_facebook_product_catalog_id',      function() { return 1; } );
  add_filter( 'wc_facebook_connection_access_token', function($access_token, $t) { return 'true'; }, 1, 2 );
```

... the plugin still thinks I am not connected. As a workaround, while we don't work on that, what I am doing to bypass and test the plugin is commenting [this line](https://github.com/facebookincubator/facebook-for-woocommerce/blob/c88c9e76bd75bf5ee004273d7b529c09dd986f44/includes/Admin.php#L48). After that I'm able to see the modal.

image

**Comment by adrianostanley** [comment]
@deivamagalhaes here's what I noticed while QAing this PR:

1. The refund modal will show even if the Order already has the Refund status. Is that correct?
2. When I choose a refund reason and click on Submit refund, another modal is shown (this one looks like the browser's default one). But the page is being submitted at the same time so this second modal won't do anything.

The video capture below shows both issues.

4KLUWOl00P

File: includes/Admin/Orders.php [hunk 1]
```diff
@@ -323,15 +323,18 @@ private function get_cancel_modal_buttons() {
...
-	public function render_refund_reason_field() {
+	public function render_refund_reason_field( $select_id = 'wc_facebook_refund_reason', $hidden = true ) {
 
 		if ( ! $this->is_edit_order_screen() ) {
 			return;
 		}
 
 		?>
-		<select id="wc_facebook_refund_reason" style="display: none;">
+		<select id="<?php echo $select_id; ?>" <?php echo $hidden ? 'style="display: none;"' : ''; ?>>
```
Comment (adrianostanley) [inline includes/Admin/Orders.php:337]:
```suggestion
		<select id="<?php echo esc_attr( $select_id ); ?>" <?php echo $hidden ? 'style="display: none;"' : ''; ?>>
```
Not a big deal, but for the sake of safety.

File: assets/js/admin/orders.js [hunk 2]
```diff
@@ -11,4 +11,51 @@ jQuery( document ).ready( ( $ ) => {
...
+	}
+
+
+	$form.on( 'submit', function( event ) {
+
+		if ( ! isCommerceOrder || $form.data('allow-submit') ) {
+			return;
+		}
+
+		if ( 'wc-refunded' === $orderStatusField.val() ) {
```
Comment (adrianostanley) [inline assets/js/admin/orders.js:56]: Unless a refunded order can be refunded again (which is a question I also added to this review), I guess you need to check the original status - like you did here https://github.com/facebookincubator/facebook-for-woocommerce/pull/1592/commits/9997deb7b221d6b6551c05d44d5266e185c8140d.

**Review by adrianostanley — CHANGES_REQUESTED** [review]

File: assets/js/admin/orders.js (hunk 2 above)
Comment (adrianostanley) [reply assets/js/admin/orders.js]: Done in https://github.com/facebookincubator/facebook-for-woocommerce/pull/1592/commits/5c1a5fdf677675269db0f5a24f93f5dcaec0f224.

**Comment by adrianostanley** [comment]
> @deivamagalhaes here's what I noticed while QAing this PR:
> 
> 1. The refund modal will show even if the Order already has the Refund status. Is that correct?
> 2. When I choose a refund reason and click on Submit refund, another modal is shown (this one looks like the browser's default one). But the page is being submitted at the same time so this second modal won't do anything.
> 
> The video capture below shows both issues.
> 
> 4KLUWOl00P

Fixed in https://github.com/facebookincubator/facebook-for-woocommerce/pull/1592/commits/1a010b39c7258ba63fe06f1878fea4061853c5f6.

**Pull request merged**